src/models/applicationState.ts

```diff
@@ -79,6 +79,16 @@ export interface IConnection {
     providerOptions: object;
 }
 
+export interface ICloudStorageOptions {
```

Comment:
Different types of cloud storage may require different settings...   I think if we introduce the enum to designate a provider as a "cloud" provider that should be good enough.